Package.swift

```diff
@@ -29,7 +29,6 @@ var targets: [Target] = [
         name: "TuistCore",
         dependencies: [
             pathDependency,
-            "ProjectDescription",
```

Comment:
The Tuist App depends on `TuistCore` and `TuistSupport`. If we kept the `ProjectDescription`, the Tuist App would have to unnecessarily embed the `ProjectDescription`. The usages of `ProjectDescription` in these targets were minimal, so I moved the necessary code to downstream targets.